File: Array/167.TwoSumII.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // common problem: using two pointer
        int left = 0, right = numbers.size() - 1;
        int s = numbers[left] + numbers[right];
        while (s != target && left < right){
            if (s > target){
                s -= numbers[right] - numbers[right-1];
                right--;
            }else {
                s += numbers[left+1] - numbers[left];
                left++;
            }
        }
        // for answer
        return {left+1, right+1};
    }
};

File: Array/167.TwoSumII.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1
        s = numbers[l] + numbers[r]
        while (s != target and l < r):
            if s > target:
                s -= numbers[r] - numbers[r-1]
                r -= 1
            else:
                s += numbers[l+1] - numbers[l]
                l +=1
        return [l+1, r+1]

        